src/main/java/org/betterx/betterend/entity/SilkMothEntity.java

```diff
@@ -76,11 +81,12 @@ public void setHive(Level world, BlockPos hive) {
 
     @Override
     public boolean canBeLeashed(Player player) {
-        return false;
+        return true;
     }
 
     @Override
     public void addAdditionalSaveData(CompoundTag tag) {
+        super.addAdditionalSaveData(tag);
```

Comment:
I'm not sure if it was intentional not to call the super method, but *I believe* that without this, not even custom names would be saved (see: despawnable).